File: backend/api/schemas/health.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class HealthStatus(str, Enum):
    """Health status values"""
    HEALTHY = "healthy"
    DEGRADED = "degraded"
    UNHEALTHY = "unhealthy"


class ComponentHealth(BaseModel):
    """Health status of a single component"""
    name: str
    status: HealthStatus
    response_time_ms: float | None = None
    message: str | None = None
    details: dict | None = None


class HealthResponse(BaseModel):
    """Complete health check response"""
    status: HealthStatus
    timestamp: datetime = Field(default_factory=datetime.now)
    version: str = "2.0.0"
    uptime_seconds: float = 0
    components: list[ComponentHealth] = []

    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat()
        }


class ReadinessResponse(BaseModel):
    """Kubernetes readiness probe response"""
    ready: bool
    reason: str | None = None


class LivenessResponse(BaseModel):
    """Kubernetes liveness probe response"""
    alive: bool

File: frontend/src/lib/api/health.ts
import { apiClient } from './client';
import type { HealthResponse } from '@/types/api';

// Health API
export const healthApi = {
  check: async (): Promise<HealthResponse> => {
    const response = await apiClient.get('/api/v1/health');
    return response.data;
  },

  checkDetailed: async (): Promise<HealthResponse> => {
    const response = await apiClient.get('/api/v1/health/detailed');
    return response.data;
  },

  checkReadiness: async (): Promise<{ ready: boolean; reason?: string }> => {
    const response = await apiClient.get('/api/v1/health/ready');
    return response.data;
  },

  checkLiveness: async (): Promise<{ alive: boolean }> => {
    const response = await apiClient.get('/api/v1/health/live');
    return response.data;
  },

  checkComponent: async (componentName: string): Promise<{
    name: string;
    status: string;
    response_time_ms: number;
    details: Record<string, unknown>;
  }> => {
    const response = await apiClient.get(`/api/v1/health/components/${componentName}`);
    return response.data;
  },
};
